src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -514,6 +529,31 @@ private int WriteRunToErrorList(Run run, string logFilePath)
             return sarifErrors.Count;
         }
 
+        private void ShowFilteredSuppressionStateColumn()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            // Creating this table source adds "Suppression State" to the list of available columns.
+            SarifSuppressedResultsTableDataSource _ = SarifSuppressedResultsTableDataSource.Instance;
```

Comment:
>     SarifSuppressedResultsTableDataSource _ = SarifSuppressedResultsTableDataSource.Instance; [](start = 7, length = 94)

Why not create a static method to create the table source (if needed) rather than using discard?

Reply:
I see your point. The `Instance` property creates an instance of `SarifSuppressesResultsTableDataSource`, and that class's ctor _registers_ it with the table manager. I could instead just define `SarifSuppressesResultsTableDataSource.Register`, which would create the instance (again, registering the source), but would not return it. Ok, will do.

---
In reply to: [538919013](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538919013) [](ancestors = 538919013)